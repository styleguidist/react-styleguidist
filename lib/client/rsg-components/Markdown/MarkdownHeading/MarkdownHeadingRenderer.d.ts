import React from 'react';
import { JssInjectedProps } from 'rsg-components/Styled';
interface MarkdownHeadingProps extends JssInjectedProps {
    children: React.ReactNode;
    level: number;
    id?: string;
}
declare const _default: React.ComponentType<Pick<MarkdownHeadingProps, "children" | "id" | "level">>;
export default _default;
