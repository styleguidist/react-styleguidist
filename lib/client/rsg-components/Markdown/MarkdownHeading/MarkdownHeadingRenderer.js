import React from 'react';
import PropTypes from 'prop-types';
import Styled from 'rsg-components/Styled';
import Heading from 'rsg-components/Heading';

var styles = function styles(_ref) {
  var space = _ref.space;
  return {
    spacing: {
      marginBottom: space[2]
    }
  };
};

var MarkdownHeadingRenderer = function MarkdownHeadingRenderer(_ref2) {
  var classes = _ref2.classes,
      level = _ref2.level,
      children = _ref2.children,
      id = _ref2.id;
  return /*#__PURE__*/React.createElement("div", {
    className: classes.spacing
  }, /*#__PURE__*/React.createElement(Heading, {
    level: level,
    id: id
  }, children));
};

MarkdownHeadingRenderer.propTypes = {
  classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
  level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]).isRequired,
  children: PropTypes.node,
  id: PropTypes.string
};
export default Styled(styles)(MarkdownHeadingRenderer);