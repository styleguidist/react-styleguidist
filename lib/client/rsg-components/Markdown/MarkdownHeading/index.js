export { default } from 'rsg-components/Markdown/MarkdownHeading/MarkdownHeadingRenderer';