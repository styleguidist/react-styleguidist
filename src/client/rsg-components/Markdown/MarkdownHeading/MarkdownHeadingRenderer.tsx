import React from 'react';
import PropTypes from 'prop-types';
import Styled, { JssInjectedProps } from 'rsg-components/Styled';
import Heading from 'rsg-components/Heading';
import * as Rsg from '../../../../typings';

const styles = ({ space }: Rsg.Theme) => ({
	spacing: {
		marginBottom: space[2],
	},
});

interface MarkdownHeadingProps extends JssInjectedProps {
	children: React.ReactNode;
	level: number;
	id?: string;
}

const MarkdownHeadingRenderer: React.FunctionComponent<MarkdownHeadingProps> = ({
	classes,
	level,
	children,
	id,
}) => {
	return (
		<div className={classes.spacing}>
			<Heading level={level} id={id}>
				{children}
			</Heading>
		</div>
	);
};

MarkdownHeadingRenderer.propTypes = {
	classes: PropTypes.objectOf(PropTypes.string.isRequired).isRequired,
	level: PropTypes.oneOf([1, 2, 3, 4, 5, 6]).isRequired,
	children: PropTypes.any,
	id: PropTypes.string,
};

export default Styled<MarkdownHeadingProps>(styles)(MarkdownHeadingRenderer);
