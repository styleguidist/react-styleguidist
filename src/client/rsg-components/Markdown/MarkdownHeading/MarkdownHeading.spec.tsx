import React from 'react';
import renderer from 'react-test-renderer';
import MarkdownHeading from './index';

describe('Markdown Heading', () => {
	it('should render a heading with a wrapper that provides margin and an id', () => {
		const actual = renderer.create(
			<MarkdownHeading id="the-markdown-heading" level={2}>
				The markdown heading
			</MarkdownHeading>
		);

		expect(actual).toMatchSnapshot();
	});
});
