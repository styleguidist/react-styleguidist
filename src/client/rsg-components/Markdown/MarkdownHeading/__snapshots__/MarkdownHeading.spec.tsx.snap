// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Markdown Heading should render a heading with a wrapper that provides margin and an id 1`] = `
<div
  className="rsg--spacing-0"
>
  <h2
    className="rsg--heading-2 rsg--heading2-4"
    id="the-markdown-heading"
  >
    The markdown heading
  </h2>
</div>
`;
